src/debuggerCommunicationServer.ts

```diff
@@ -36,16 +55,22 @@ export class DebuggerCommunicationServer {
         this.simulatorWebview = webviewPanel;
     }
 
-    // Emit Buttons Inputs Events
-    public emitButtonPress(newState: string): void {
-        console.log(`Emit Button Press: ${newState} \n`);
-        this.serverIo.emit("button_press", newState);
-    }
-
-    // Emit Sensors Inputs Events
-    public emitSensorChanged(newState: string): void {
-        console.log(`Emit Sensor Changed: ${newState} \n`);
-        this.serverIo.emit("sensor_changed", newState);
+    public emitInputChanged(newState: string): void {
+        if (this.isWaitingResponse) {
+            this.currentCall.push(() => {
+                this.serverIo.emit(
```

Comment:
I would add a comment here explaining that since we already have an active event, we will just push this to our queue until that event is processed.